baybe/utils/scaling.py

```diff
@@ -2,21 +2,57 @@
 
 from __future__ import annotations
 
-from typing import Protocol
-
-import pandas as pd
-
-
-class ParameterScalerProtocol(Protocol):
-    """Type protocol specifying the interface parameter scalers need to implement.
-
-    The protocol is compatible with sklearn scalers such as
-    :class:`sklearn.preprocessing.MinMaxScaler` or
-    :class:`sklearn.preprocessing.MaxAbsScaler`.
-    """
-
-    def fit(self, df: pd.DataFrame, /) -> None:
-        """Fit the scaler to a given dataframe containing parameter configurations."""
-
-    def transform(self, df: pd.DataFrame, /) -> pd.DataFrame:
-        """Transform a parameter dataframe using the fitted scaling logic."""
+import itertools
+from typing import TYPE_CHECKING
+
+from attrs import define, field
+from attrs.validators import deep_iterable, deep_mapping, instance_of
+
+if TYPE_CHECKING:
+    from botorch.models.transforms.input import InputTransform
+    from torch import Tensor
+
+
+@define
+class ColumnTransformer:
```

Comment:
@Scienfitz, I can already sense that you are going to ask why I moved again away from the sklearn logic. To give you a quick heads up: I came to the conclusion, that scaling **needs** to happen after the layer where botorch connects to the posterior computation – simply because when you call `optimize_*`, you want to pass objects in their *unscaled* i.e. *raw* computational representation to it. If you did it the other way, it would always mean that outside objects need to be aware of how the surrogate internally applies scaling so that they can then prepare the passed objects on the same scale.

First problem: you'll always have to ask yourself the question "in which representation do I now need to pass stuff, scaled or not!?" and, in fact, we all fell into that trap in the last PR. So it's much better to completely hide away the scaling inside the surrogate, since it it really a surrogate-internal and model-specific decision that he outer world needs nothing to know about.

Second problem: If you do scaling outside on the numpy layer, it can never be part of the computational graph. Right now this is not a problem per se (it could become one in the future though, in case we need gradients of the objective with respect to the input), but I think this already hints us toward the fact that something is not ideal here.

Third problem: there is another reason to use the available botorch scalers, namely then can do batch-wise scaling (i.e. everything in gyptorch/botorch is set up in a way so that you can pass different slices of a tensor into different models, each with their own parameters/scaling etc). 

If we acknowledge all these issues + the "optimize_*" problem I outlined on top, the only logical conclusion is (I think) to trigger scaling behind the botorch connection point, that is on the tensor layer of the computational representation. 

I hope that makes my motivation clear. If not, please don't hesitate to reach out 👍🏼

Reply:
ok fine for me, I think sklearn interface would have been a bit easier for users and has more choices (like eg also a `RobustScaler`) but you also have some points + the integration seems to be fine now

My main worry is that some places, in particular outside of the core code, that implicitly used / reply on doing something / providing the comp rep have been overlooked. One such example I suspect is `.get_integration_points` of the `qNIPV` and custom model examples